src/transactions/feed/queryHelper.ts

```diff
@@ -101,10 +104,15 @@ export function useFetchTransactions(): QueryHookResult {
 
       if (isPollResult && returnedTransactions.length) {
         // We store the first page in redux to show them to the users when they open the app.
+        // Filter out now empty transactions to avoid redux issues
         const nonEmptyTransactions = returnedTransactions.filter(
           (returnedTransaction) => !isEmpty(returnedTransaction)
         )
-        dispatch(updateTransactions(nonEmptyTransactions))
+        // If there are new transactions update transactions in redux and fetch balances
+        if (nonEmptyTransactions[0].transactionHash !== lastTransactionHash) {
+          dispatch(updateTransactions(nonEmptyTransactions))
+          dispatch(fetchTokenBalances({ showLoading: false }))
+        }
```

Comment:
`nonEmptyTransactions` could be empty.

And we should avoid dispatching multiple actions sequentially.
We could handle it in [a saga](https://github.com/valora-inc/wallet/blob/ceb65a000bdaaa0e89b3120d2bbcfeb441ef416a/src/transactions/saga.ts#L246-L251), listening for the `updateTransactions` event and comparing the last transaction hash before and after.
This way the balance update logic would be applied whenever the action is dispatched.
Of course today this action is only dispatched here, but this makes it a bit cleaner and more resilient to future changes.

[This is related to modelling actions as events which is strongly recommended for Redux](https://redux.js.org/style-guide/#model-actions-as-events-not-setters).

Reply:
@jeanregisser would something like this work instead?
```
// If there are new transactions update transactions in redux and fetch balances
if (nonEmptyTransactions.length && nonEmptyTransactions[0].transactionHash !== lastTransactionHash) {
  batch(() => {
    dispatch(updateTransactions(nonEmptyTransactions))
    dispatch(fetchTokenBalances({ showLoading: false }))
  })
}
```

Reply:
Yes it could be a quick way (if the `updateTransaction` is never going to be used anywhere else). But no need to batch.
I think we still want to dispatch even if it's empty. And only do the state update if it changes.

Reply:
i feel strange about `batch` but only because we don't use it anywhere else and i prefer to use existing patterns where we can to keep things predictable. however i do think that using `batch` simplifies the implementation a lot and reduces the number of times we execute on the reducer right? 

i'm thinking that our choices are either to:
- dispatch an action on every poll (without any reducers listening for this action) -> have a saga that checks for new tx's and if there are new ones dispatch these 2 actions (seems like the same issue of dispatching multiple actions sequentially)
- dispatch an action on every poll -> the redux store is updated every time -> have a saga that refreshes the token balance on this action. idk how much of a performance impact it is to hit the redux store every 10 seconds, but i would have thought to avoid this if we can.
- create a new action that is dispatched when there are new tx's -> this action is the combination of `updateTransactions` and `fetchTokenBalances` -> have the saga that currently responds to `fetchTokenBalances` also respond to this new action

is there are more optimal solution involving sagas that i'm missing? i think the last option is nice but not totally convinced that it's nicer than what we have now since it's more complicated 🙈